File: 프로그래머스/2/62048. 멀쩡한 사각형/멀쩡한 사각형.cpp
#include <cmath>

using namespace std;

long long solution(int w,int h) {
    if (w == h) return (long long)w * (w -1);
    if (w == 1 || h == 1) return 0;
    long long cnt = 0;
    for (int i = 1; i <= w; i++)
        cnt += ceil((double)i * h / w) - floor((double)(i-1) * h / w);
    return (long long)w * h - cnt;
}

File: 프로그래머스/2/62048. 멀쩡한 사각형/멀쩡한 사각형.java
class Solution {
    public long solution(int w, int h) {
        double lin = (double) h / w;
        long count = (long) w * h, l, r;
        for (int i = 1; i <= w; i++) {
            l = (long)Math.floor((double) h * (i-1) / w);
            r = (long)Math.ceil((double) h * i / w);
            count -= r - l;
        }
        return count;
    }
}